src/SLCore.UnitTests/State/ActiveConfigScopeTrackerTests.cs

```diff
@@ -0,0 +1,229 @@
+        var testSubject = CreateTestSubject(serviceProvider.Object, lockFactory.Object, threadHandling.Object);
+        testSubject.currentConfigScope = new ConfigurationScopeDto(configScopeId, configScopeId, true, null);
+
+        var currentScope = testSubject.Current;
+
+        currentScope.Should().BeEquivalentTo(new ConfigurationScope(configScopeId));
+        VerifyThreadHandling(threadHandling);
+        VerifyLockTakenSynchronouslyAndReleased(asyncLock, lockRelease);
+    }
+    
+    [TestMethod]
+    public void GetCurrent_ReturnsBoundScope()
+    {
+        const string configScopeId = "myid";
+        const string connectionId = "myconid";
+        const string sonarProjectKey = "projectkey";
+        var threadHandling = new Mock<IThreadHandling>();
+        ConfigureServiceProvider(out var serviceProvider, out _);
+        ConfigureAsyncLockFactory(out var lockFactory, out var asyncLock, out var lockRelease);
+        var testSubject = CreateTestSubject(serviceProvider.Object, lockFactory.Object, threadHandling.Object);
+        testSubject.currentConfigScope = new ConfigurationScopeDto(configScopeId, configScopeId, true, new BindingConfigurationDto(connectionId, sonarProjectKey));
+
+        var currentScope = testSubject.Current;
+
+        currentScope.Should().BeEquivalentTo(new ConfigurationScope(configScopeId, connectionId, sonarProjectKey));
+        VerifyThreadHandling(threadHandling);
+        VerifyLockTakenSynchronouslyAndReleased(asyncLock, lockRelease);
+    }
+    
+    [TestMethod]
+    public void Dispose_DisposesLock()
+    {
+        ConfigureServiceProvider(out var serviceProvider, out _);
+        ConfigureAsyncLockFactory(out var lockFactory, out var asyncLock, out _);
+        
+        var testSubject = CreateTestSubject(serviceProvider.Object, lockFactory.Object, Mock.Of<IThreadHandling>());
+        
+        testSubject.Dispose();
+        asyncLock.Verify(x => x.Dispose());
+    }
+
+    private static void VerifyThreadHandling(Mock<IThreadHandling> threadHandling)
+    {
+        threadHandling.Verify(x => x.ThrowIfOnUIThread());
+    }
+
+    private static void VerifyServiceAddCall(Mock<IConfigurationScopeSLCoreService> configScopeService, ActiveConfigScopeTracker testSubject)
+    {
+        configScopeService
+            .Verify(x =>
+                    x.DidAddConfigurationScopesAsync(It.Is<DidAddConfigurationScopesParams>(p =>
+                        p.addedScopes.SequenceEqual(new[] { testSubject.currentConfigScope }))),
+                Times.Once);
+    }
+
+    private static void VerifyLockTakenSynchronouslyAndReleased(Mock<IAsyncLock> asyncLock, Mock<IReleaseAsyncLock> lockRelease)
+    {
+        asyncLock.Verify(x => x.Acquire(), Times.Once);
+        lockRelease.Verify(x => x.Dispose(), Times.Once);
+    }    
+    
+    private static void VerifyLockTakenAndReleased(Mock<IAsyncLock> asyncLock, Mock<IReleaseAsyncLock> lockRelease)
```

Comment:
maybe use a mocksequence to make sure they're called in correct order

Reply:
Acquire returns the releaser (as per `ConfigureAsyncLockFactory`), so there's no need to check that.